core/src/main/java/org/apache/iceberg/rest/http/RequestResponseSerializers.java

```diff
@@ -0,0 +1,162 @@
+    module
+        .addSerializer(TableIdentifier.class, new RequestResponseSerializers.TableIdentifierSerializer())
+        .addDeserializer(TableIdentifier.class, new RequestResponseSerializers.TableIdentifierDeserializer())
+        .addSerializer(Namespace.class, new RequestResponseSerializers.NamespaceSerializer())
+        .addDeserializer(Namespace.class, new RequestResponseSerializers.NamespaceDeserializer())
+        .addSerializer(Schema.class, new RequestResponseSerializers.SchemaSerializer())
+        .addDeserializer(Schema.class, new RequestResponseSerializers.SchemaDeserializer())
+        .addSerializer(PartitionSpec.class, new RequestResponseSerializers.PartitionSpecSerializer())
+        .addDeserializer(PartitionSpec.class, new RequestResponseSerializers.PartitionSpecDeserializer())
+        .addSerializer(SortOrder.class, new RequestResponseSerializers.SortOrderSerializer())
+        .addDeserializer(SortOrder.class, new RequestResponseSerializers.SortOrderDeserializer())
+        .addSerializer(TableMetadata.class, new RequestResponseSerializers.TableMetadataSerializer());
+    mapper.registerModule(module);
+  }
+
+  public static class NamespaceDeserializer extends JsonDeserializer<Namespace> {
+    @Override
+    public Namespace deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return Namespace.of(jsonNode.asText().split("\\."));
+    }
+  }
+
+  public static class NamespaceSerializer extends JsonSerializer<Namespace> {
+    @Override
+    public void serialize(Namespace namespace, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+      gen.writeString(namespace.toString());
+    }
+  }
+
+  public static class TableIdentifierDeserializer extends JsonDeserializer<TableIdentifier> {
+    @Override
+    public TableIdentifier deserialize(JsonParser p, DeserializationContext ctxt) throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      return TableIdentifier.parse(jsonNode.asText());
+    }
+  }
+
+  public static class TableIdentifierSerializer extends JsonSerializer<TableIdentifier> {
+    @Override
+    public void serialize(TableIdentifier identifier, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
+      gen.writeString(identifier.toString());
+    }
+  }
+
+  public static class SchemaDeserializer extends JsonDeserializer<Schema> {
+    @Override
+    public Schema deserialize(JsonParser p, DeserializationContext context)
+        throws IOException {
+      JsonNode jsonNode = p.getCodec().readTree(p);
+      Schema schema = SchemaParser.fromJson(jsonNode);
+      context.setAttribute("schema", schema);
+      return schema;
+    }
+  }
+
+  public static class SchemaSerializer extends JsonSerializer<Schema> {
+    @Override
+    public void serialize(Schema schema, JsonGenerator gen, SerializerProvider serializers)
+        throws IOException {
```

Comment:
Does this actually throw `IOException`? If not, can we remove it?

Reply:
Yeah this does throw `IOException`. Specifically, `SchemaParser.toJson` does.

I can wrap it with the `UncheckedIOException` or something else though.

Reply:
Actually, my mistake. The `serialize` method itself declares the `IOException`. It _looks like_ there's a delegate that can be used, so either I'll figure out the existing delegate or wrap with my own to avoid all of the checked exceptions if possible.